lib/fs.js

```diff
@@ -609,6 +609,12 @@ fs.read = function(fd, buffer, offset, length, position, callback) {
     };
   }
 
+  if (length === 0) {
+    return process.nextTick(function() {
```

Comment:
You comment prompted me to check the `fs.readSync` case and the issue you describe exists there. I'll fix it and add a test.